token_dispenser/token_dispenser_lambda.py

```diff
@@ -0,0 +1,161 @@
+import token_dispenser.configuration as config
+from token_dispenser.repository.client_token import put_token, get_token_by_client_id
+from token_dispenser.repository.client_token import ClientTokens
+from token_dispenser.logger import TokenDispenserLogger
+import json
+import logging
+import time
+
+def decode_pkcs12(p12_file_path, password: str=None):
+    """
+    Decodes a PKCS#12 file to extract the private key, certificate, and additional certificates.
+
+    :param p12_file_path: Path to the PKCS#12 file (.p12 or .pfx).
+    :param password: Password for the PKCS#12 file (if any).
+    :return: A tuple containing the private key, certificate, and additional certificates.
+    """
+    with open(p12_file_path, "rb") as file:
+        pkcs12_data = file.read()
+
+    # Decode the PKCS#12 file
+    private_key, certificate, additional_certs = load_key_and_certificates(
+        pkcs12_data,
+        password.encode("utf-8") if password else None,
+        backend=default_backend()
+    )
+    return private_key, certificate, additional_certs
+
+
+def get_new_token(client_id:str):
+    """
+    Decodes a PKCS#12 file to extract the private key, certificate, and additional certificates.
+
+    :param client_id: a required field which is used as key to cache the token in dynamoDB.
+    :return: a json object containing the token data.
+    """
+    try:
+        logger = TokenDispenserLogger.get_logger()
+        secret_manager_util = SecretManager()
+        # p12_file = secret_manager_util.get_secret_binary(config.LAUNCHPAD_PFX_BINARY_SECRET_KEY)
+        s3_util = S3()
+        p12_file = s3_util.download_s3_file(bucket_name=config.LAUNCHPAD_PFX_FILE_S3_BUCKET,
+                                            key=config.LAUNCHPAD_PFX_FILE_S3_KEY, local_storage_dir='/tmp')
+        logger.info(f"p12 file downloaded from s3 successfully to: {p12_file}")
+        password = secret_manager_util.get_secret_value(config.LAUNCHPAD_PFX_PASSWORD_SECRET_ID)
+        private_key, cert, additional_certs = decode_pkcs12(p12_file, password)
+        logger.info(f"cert files decoded successfully")
+        # Create launchpad token
+        launchpad_token_util = LaunchpadToken()
+        token_json = launchpad_token_util.get_token(url=config.LAUNCHPAD_GETTOKEN_URL, private_key=private_key,
+                                                        certificate=cert)
+        # Add created_at and expires_at fields into the token_structure before putting into dynamoDB
+        current_time:int=int(time.time())
+        token_json['expires_at'] = current_time + int(token_json['session_maxtimeout'])
+        token_json['created_at'] = current_time
+        put_token(client_id, json.dumps(token_json), (time.time() + config.CLIENT_EXPIRATION_TIME))
+        return token_json
+    except Exception as e:
+        print(f"Failed on launchpad token process: {e}")
+        raise e
+
+
+def satisfy_minimum_alive_secs(token_json:json, minimum_alive_secs:int | None) -> bool:
```

Comment:
minimum_alive_secs should never be None because it defaults to DEFAULT_TOKEN_MIN_ALIVE_TIME in all the function calls. Let's not make that parameter optional